cli/cmd/repair.go

```diff
@@ -43,6 +47,31 @@ linkerd upgrade.`,
 				return errors.New("secret/linkerd-config-overrides already exists. If you need to regenerate this resource, please delete it before proceeding")
 			}
 
+			// Check if the CLI version matches with that of the server
+			clientVersion := version.Version
+			var serverVersion string
+			publicClient, err := public.NewExternalClient(cmd.Context(), controlPlaneNamespace, k8sAPI)
+			if err != nil {
+				return err
+			}
+			serverVersion, err = healthcheck.GetServerVersion(cmd.Context(), publicClient)
+			if err != nil {
+				return err
+			}
+
+			if serverVersion != clientVersion {
+				versionRegex, err := regexp.Compile("stable-2.[8-9].*")
```

Comment:
Consider using MustCompile and putting this regex in a global rather than a local.